File: insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/app_layer/parameter_blocks/TBROverNotificationBlock.java
package info.nightscout.androidaps.plugins.pump.insight.app_layer.parameter_blocks;

import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf;

public class TBROverNotificationBlock extends ParameterBlock {

    private boolean enabled;
    private int melody;

    @Override
    public void parse(ByteBuf byteBuf) {
        enabled = byteBuf.readBoolean();
        melody = byteBuf.readUInt16LE();
    }

    @Override
    public ByteBuf getData() {
        ByteBuf byteBuf = new ByteBuf(4);
        byteBuf.putBoolean(enabled);
        byteBuf.putUInt16LE(melody);
        return byteBuf;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }
}

File: insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/app_layer/parameter_blocks/TBROverNotificationBlock.kt
package info.nightscout.androidaps.plugins.pump.insight.app_layer.parameter_blocks

import info.nightscout.androidaps.plugins.pump.insight.utils.ByteBuf

class TBROverNotificationBlock : ParameterBlock() {

    internal var isEnabled = false
    private var melody = 0

    override fun parse(byteBuf: ByteBuf) {
        isEnabled = byteBuf.readBoolean()
        melody = byteBuf.readUInt16LE()
    }

    override val data: ByteBuf
        get() {
            val byteBuf = ByteBuf(4)
            byteBuf.putBoolean(isEnabled)
            byteBuf.putUInt16LE(melody)
            return byteBuf
        }
}